src/components/Item.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import db from '../lib/firebase';
+
+const Item = ({ name, id, lastPurchasedDate }) => {
+  const checkHandler = () => {
+    db.collection('items').doc(id).update({ lastPurchasedDate: new Date() });
+  };
```

Comment:
Yea I noticed what Kevin talked about too. It looks like you need to adjust this function a little to check if the item had been checked and updated in the db initially. I have a suggestion on how you can do that.

1) an if statement to check if lastPurchasedDate is null, if true, update the db with current time,
2) in the else block, return the function, that way it won't do anything if lastpurchasedDate is not null.

it works for me locally, let me know if you have any questions about the approach should you choose it.

Reply:
Thank you so much @Amaka202 , This gonna help a lot.

Reply:
Yes, absolutely agree here 👆. This is a great suggestion. We need to control the `onChange` event handler after we make the first purchase (when we check the input box). We should totally disable the checkbox for 24 hours because the same input field (`checkbox`) will help us (in the coming week's issue) to track the `numberOfPuchases` the user makes.